event-adapter/cmd/provider/translator.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	sd "google.golang.org/api/logging/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api "k8s.io/client-go/pkg/api/v1"
+)
+
+type eventTime struct {
```

Comment:
What is the benefit of defining new struct? Can you use metav1.Time instead?

Reply:
I use this struct in order to decode the response. I need to access the field timestamp in the response, in this way it decodes the response in an eventTime struct and put the timestamp value in the field Timestamp. I do not know if there is a better way that allow me to access that value.

Reply:
Removed the struct, used a map instead.